### ISSUE TITLE
Update code to most recent o1js for Tutorial 10: Account Updates

### ISSUE THREAD

**Issue by jackryanservia** [MinaProtocol/docs2#660]
The example code for Tutorial 10: Account Updates needs to be updated so it's compatible with current versions of o1js.
More code focus required:
Feedback from @qwadratic for [Tutorial 10: Tutorial 10: Account Updates >](https://docs.minaprotocol.com/zkapps/tutorials/account-updates#non-upgradeable-proof-only) Non-upgradeable proof only: it is stated that asserting provedState == false is recommended in zkApps (multiple times), however we don’t use this assertion anywhere else.

**Comment by barriebyron** [comment]
@ymekuria  yes... can you help me understand the question? (I was wondering if this issue were in another repo... but it looks to me like docs2 is correct)

**Comment by Pfed-prog** [comment]
I think I had a very similar though when I went through tutorial 10. It says that 
![image](https://github.com/o1-labs/docs2/assets/66903336/8597157e-ad7d-43e1-8667-f6c051508aa8)

But in all other code examples this pattern is absent.

**Comment by Pfed-prog** [comment]
My understanding is that the following function runs on deploy and sets this.account.provedState to true

  ```ts
@method init() {
    this.account.provedState.requireEquals(this.account.provedState.get());
    this.account.provedState.get().assertFalse();

    super.init();
    this.num.set(Field(1));
    this.calls.set(Field(0));
  }
```

**Comment by dkijania** [comment]
Fixed by #1252, merged yesterday.

The tutorial 10 example is on o1js `^3.0.0` with a lockfile that resolves it, and it is no longer only built: `node build/src/main.js` runs in CI on Node 22 and 24 on every pull request, and a daily job builds it against `o1js@latest` so the next release surfaces as a failing scheduled run rather than as a stale tutorial.

For context on how stale it had become: seven packages were pinned to a peer range of `^2.0.0` and four to 1.x, while npm was shipping 3.0.0.
